plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -225,6 +231,57 @@ private DeltaLakeColumnHandle buildColumnHandle(EntryType entryType, CheckpointS
         return new DeltaLakeColumnHandle(entryType.getColumnName(), type, OptionalInt.empty(), entryType.getColumnName(), type, REGULAR, Optional.empty());
     }
 
+    /**
+     * Constructs a TupleDomain which filters on a specific required primitive sub-column of the EntryType being
+     * not null for effectively pushing down the predicate to the Parquet reader.
+     */
+    private TupleDomain<HiveColumnHandle> buildTupleDomainColumnHandle(EntryType entryType, HiveColumnHandle column)
+    {
+        String field;
+        Type type;
+        switch (entryType) {
+            case TRANSACTION:
```

Comment:
You could combine some of these cases.
```suggestion
            case COMMIT:
            case TRANSACTION:
```

Same deal with ADD and REMOVE